src/unix.rs

```diff
@@ -399,6 +400,80 @@ impl ThreadPriority {
     }
 }
 
+// On apple targets, only `pthread_setschedparam` is allowed. `nice` will fail when increasing the priority if the use is not superuser.
+#[cfg(any(target_os = "macos", target_os = "ios"))]
+impl ThreadPriority {
+    /// Returns the maximum allowed value for using with the provided policy.
+    /// The returned number is in the range of allowed values.
+    pub fn max_value_for_policy(policy: ThreadSchedulePolicy) -> Result<libc::c_int, Error> {
+        let max_priority = unsafe { libc::sched_get_priority_max(policy.to_posix()) };
+        if max_priority < 0 {
+            Err(Error::OS(errno()))
+        } else {
+            Ok(max_priority)
+        }
+    }
+
+    /// Returns the minimum allowed value for using with the provided policy.
+    /// The returned number is in the range of allowed values.
+    pub fn min_value_for_policy(policy: ThreadSchedulePolicy) -> Result<libc::c_int, Error> {
+        let min_priority = unsafe { libc::sched_get_priority_min(policy.to_posix()) };
+        if min_priority < 0 {
+            Err(Error::OS(errno()))
+        } else {
+            Ok(min_priority)
+        }
+    }
+
+    /// Checks that the passed priority value is within the range of allowed values for using with the provided policy.
+    pub fn to_allowed_value_for_policy(
```

Comment:
This is a complete duplicate of the original `to_allowed_value_for_policy`, I don't think we should duplicate the code. If it can be reused, it should be.